ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/AtomicActionRecoveryModule.java

```diff
@@ -167,6 +170,28 @@ private boolean isTransactionInMidFlight( int status )
       return inFlight ;
    }
 
+   private boolean isTransactionHeuristic( int status )
+   {
+       boolean heuristic = false ;
+
+       switch ( status )
+       {
+           case ActionStatus.H_COMMIT   :
+           case ActionStatus.H_MIXED    :
+           case ActionStatus.H_HAZARD   :
+           case ActionStatus.H_ROLLBACK :
+               heuristic = true ;
+               break ;
+
+           // this shouldn't happen
+           case ActionStatus.INVALID :
```

Comment:
I am a bit curious to know why you want to call out INVALID rather than just rely on default?

Reply:
This function is based on `private boolean isTransactionInMidFlight( int status )` (i.e. copy and paste)

Reply:
We can definitely rely on the first assignment of the `heuristic` variable. Modified.

Reply:
Thank you!